compiler/rustc_errors/src/emitter.rs

```diff
@@ -1376,8 +1376,17 @@ impl EmitterWriter {
                     let file = annotated_file.file.clone();
                     let line = &annotated_file.lines[line_idx];
                     if let Some(source_string) = file.get_line(line.line_index - 1) {
-                        let leading_whitespace =
-                            source_string.chars().take_while(|c| c.is_whitespace()).count();
+                        let leading_whitespace = source_string
+                            .chars()
+                            .take_while(|c| c.is_whitespace())
+                            .map(|c| {
+                                match c {
+                                    // Tabs are displayed as 4 spaces
+                                    '\t' => 4,
+                                    _ => 1,
```

Comment:
Is newlines part of this?

Reply:
No, we’re only dealing with a single source line at a time here, so we can ignore new lines.

Reply:
Should there be a comment here mentioning that?

Reply:
Hmm, the surrounding code is iterating line by line already, so to me it would be quite surprising to encounter a new line at this point. Even if it did come through, it would mean the source line is all whitespace, so it’s value will be ignored on the next line. I’d be happy to add a comment if there’s agreement it would be useful, though!

Reply:
Okay, I don't have a lot of context about this but if that is the case then I think it shouldn't be necessary.

Reply:
There are [other characters that are considered whitespace](https://doc.rust-lang.org/reference/whitespace.html) that could appear here, but I think that assuming a single char width for them is reasonable enough (don't think we can have a ZWS in there :smile:).